resources/fetching/fetchers/azure/assets_fetcher.go

```diff
@@ -0,0 +1,139 @@
+type AzureAssetInfo struct {
+	Id             string         `json:"id,omitempty"`
+	Name           string         `json:"name,omitempty"`
+	Location       string         `json:"location,omitempty"`
+	Properties     map[string]any `json:"properties,omitempty"`
+	ResourceGroup  string         `json:"resource_group,omitempty"`
+	SubscriptionId string         `json:"subscription_id,omitempty"`
+	TenantId       string         `json:"tenant_id,omitempty"`
+}
+
+var AzureAssetTypes = map[string]string{
+	"microsoft.compute/virtualmachines": fetching.AzureVMType,
+	"microsoft.storage/storageaccounts": fetching.AzureStorageAccountType,
+}
+
+func NewAzureAssetsFetcher(log *logp.Logger, ch chan fetching.ResourceInfo, provider inventory.ServiceAPI) *AzureAssetsFetcher {
+	return &AzureAssetsFetcher{
+		log:        log,
+		resourceCh: ch,
+		provider:   provider,
+	}
+}
+
+func (f *AzureAssetsFetcher) Fetch(ctx context.Context, cMetadata fetching.CycleMetadata) error {
+	f.log.Info("Starting AzureAssetsFetcher.Fetch")
+	// TODO: Maybe we should use a query per type instead of listing all assets in a single query
+	// This might be relevant if we'd like to fetch assets in parallel in order to evaluate a rule that uses multiple resources
+	assets, err := f.provider.ListAllAssetTypesByName(maps.Keys(AzureAssetTypes))
+	if err != nil {
+		return err
+	}
+
+	for _, asset := range assets {
+		select {
+		case <-ctx.Done():
+			f.log.Infof("AzureAssetsFetcher.Fetch context err: %s", ctx.Err().Error())
+			return nil
+		case f.resourceCh <- fetching.ResourceInfo{
+			CycleMetadata: cMetadata,
+			// TODO: Safe guard this conversion
+			Resource: getAssetFromData(asset.(map[string]any)),
+		}:
+		}
+	}
+
+	return nil
+}
+
+func getAssetFromData(data map[string]any) *AzureAsset {
+	assetType := getString(data, "type")
+	properties, _ := data["properties"].(map[string]any)
+
+	return &AzureAsset{
+		Type:    AzureAssetTypes[assetType],
+		SubType: getAzureSubType(assetType),
+		Asset: AzureAssetInfo{
+			Id:             getString(data, "id"),
+			Name:           getString(data, "name"),
+			Location:       getString(data, "location"),
+			Properties:     properties,
+			ResourceGroup:  getString(data, "resourceGroup"),
+			SubscriptionId: getString(data, "subscriptionId"),
+			TenantId:       getString(data, "tenantId"),
+		},
+	}
+}
+
+func getString(data map[string]any, key string) string {
+	value, _ := data[key].(string)
+	return value
+}
+
+// TODO: Handle this function
+func (f *AzureAssetsFetcher) Stop() {
+	// f.provider.Close()
+}
+
+func (r *AzureAsset) GetData() any {
+	return r.Asset
+}
+
+func (r *AzureAsset) GetMetadata() (fetching.ResourceMetadata, error) {
+	return fetching.ResourceMetadata{
+		ID:      r.Asset.Id,
+		Type:    r.Type,
+		SubType: r.SubType,
+		Name:    r.Asset.Name,
+		Region:  r.Asset.Location,
+	}, nil
+}
+
+func (r *AzureAsset) GetElasticCommonData() (map[string]any, error) { return nil, nil }
+
+// TODO: Implement this function
```

Comment:
What's the significance of this? Is it a blocker?

Reply:
It is not a blocker because we did not decide on the subTypes yet.
As of current we only utilize the resource type and do not look at subTypes in the policies.
Future implementation of additional rules might change that, this is why it is left blank as of current.